### ISSUE TITLE
can't add docker secret in all namespaces

### ISSUE THREAD

**Issue by TXZdream** [rancher/rancher#27579]
**What kind of request is this (question/bug/enhancement/feature request):**

bug

**Steps to reproduce (least amount of steps as possible):**

1. choose a project
1. add a docker secret and select all namespaces
1. finish writing and save

**Result:**

It can't be saved and the console will print 'TypeError: this.namespace is null'

**Other details that may be helpful:**

When you choose single namesapce and then choose back to all namespaces, the save will work

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.4.4
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): K3s
- Machine type (cloud/VM/metal) and specifications (CPU/memory): cloud
- Kubernetes version (use `kubectl version`): 

```
version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.3+k3s1", GitCommit:"96653e8d86b74a1f73eadf09d4edc44778c2f8dd", GitTreeState:"clean", BuildDate:"2020-06-01T16:47:11Z", GoVersion:"go1.13.11", Compiler:"gc", Platform:"linux/amd64"}
```

- containerd version (use `docker version`):

```
 1.3.3-k3s2
```

**Comment by bengray2** [comment]
I can confirm I am having the exact same issue - I can't add a private repo.   Rancher v2.4.4 with Rancher UI v2.4.22.

**Comment by ronaldgrn** [comment]
I can confirm I'm also having this issue trying to add the github registry (docker.pkg.github.com) in rancher v2.4.4. It however works if I specify a namespace.

**Comment by w3blogfr** [comment]
I think I have the same issue with Rancher 2.4.3
I think the issue is coming from here
assets/ui/components/cru-registry/component.js (line 78)

```
    willSave: function willSave() {
      var nsId = this.namespace.id,
          pr = this.primaryResource;
      Ember.set(pr, 'namespaceId', nsId ? nsId : TEMP_NAMESPACE_ID);
      var ok = this.validate();
      return ok;
    },
```

as this.namespace is null, we can create the registry.
I temporary resolve the issue by opening chrome debugger, added a breakpoint and execute this javascript command.
```
this.namespace={}
```

**Comment by w3blogfr** [comment]
In fact, I see it has been already fixed in the ticket #26981

**Comment by sowmyav27** [comment]
Issue is fixed in this this - https://github.com/rancher/rancher/issues/26981
Closing as its a duplicate.
